src/reconcile/rec_write.c

```diff
@@ -1889,15 +1889,14 @@ __wt_split_page_size(WT_BTREE *btree, uint32_t maxpagesize)
 	 * we don't waste space when we write).
 	 */
 	a = maxpagesize;			/* Don't overflow. */
-	split_size = (uint32_t)
-	    WT_ALIGN((a * (u_int)btree->split_pct) / 100, btree->allocsize);
+	split_size = (uint32_t)WT_ALIGN_NEAREST(
+	    (a * (u_int)btree->split_pct) / 100, btree->allocsize);
 
 	/*
-	 * If the result of that calculation is the same as the allocation unit
-	 * (that happens if the maximum size is the same size as an allocation
-	 * unit, use a percentage of the maximum page size).
+	 * Respect the configured split percentage if the calculated split
+	 * size is either zero or a full page.
 	 */
-	if (split_size == btree->allocsize)
+	if (split_size == 0 || split_size == maxpagesize)
 		split_size = (uint32_t)((a * (u_int)btree->split_pct) / 100);
```

Comment:
I thought we intended to allow applications to set the split percentage to 100%?

Also, I'm a little worried about the original code here: it seems to me if you set both the allocation size and the maximum page size to the minimum possible allocation size, you can end up with a calculated split_size that's less than an allocation size? 

What do you think of something like this, that allows 100%, and bounds the calculated split size more simply?

```
uint32_t
__wt_split_page_size(WT_BTREE *btree, uint32_t maxpagesize)
{
        uintmax_t a;
        uint32_t split_size;

        /*
         * The application may not be modifying records or inserting anywhere
         * but the end, allow maximum page-size splits.
         */
        if (btree->split_pct == 100)
                return (maxpagesize);

        /*
         * Ideally, the split page size is some percentage of the maximum page
         * size rounded to an allocation unit (round to an allocation unit so
         * we don't waste space when we write).
         */
        a = maxpagesize;                        /* Don't overflow. */
        split_size = (uint32_t)WT_ALIGN_NEAREST(
            (a * (u_int)btree->split_pct) / 100, btree->allocsize);

        /*
         * Bound the calculated split size at an allocation size minimum and  
         * a maximum page size minus an allocation size, maximum.
         */
        if (split_size == maxpagesize)
                split_size -= btree->allocsize;
        if (split_size < btree->allocsize)
                split_size = btree->allocsize;

        return (split_size);
}
```